docs/docs/api/functions/queue.md

```diff
@@ -9,10 +9,10 @@ added track index. If no tracks were added it returns `void`.
 | Param          | Type     | Description   |
 | -------------- | -------- | ------------- |
 | tracks         | `array` of [Track Object](../objects/track.md) or a single one | The tracks that will be added |
-| insertBeforeIndex | `number` | The index of the track that will be located immediately after the inserted tracks. Set it to `null` to add it at the end of the queue |
+| insertBeforeIndex | `number` | (Optional) The index of the track that will be located immediately after the inserted tracks. By default the track is added to the end of the queue |
 
 ## `remove(tracks)`
-Removes one or more tracks from the queue.
+Removes one or more tracks from the queue. If the current track is removed, the next track will activated. If the current track was the last track in the queue, the first track will be activated.
```

Comment:
I'm wondering, what happens when there are no tracks left in the queue? Does the player return to "Ready" state? or what?

Reply:
Good question. Currently it probably depends on the platform. On Android I am guessing `State.Stopped`, because it is linked to exoplayer's `Player.STATE_ENDED` which means `The player finished playing all media.`. On iOs it probably would be [`State.None`](https://react-native-track-player.js.org/docs/api/constants/state#:~:text=Description-,None,-State%20indicating%20that): 

> State indicating that no media is currently loaded

I created this issue to deprecate `State.Stopped` and replace it with `State.Ended` https://github.com/doublesymmetry/react-native-track-player/issues/1791

But I am also working on adding back `TrackPlayer.stop()` as per https://github.com/doublesymmetry/react-native-track-player/issues/1786 which might mean a breaking change for `State.Stopped` to mean the loading of the current active track has been stopped.

All in all it also comes down to us needing cross platform tests in place to make sure event order can be relied upon to be equal across platforms.